### PR TITLE
Query : Get Rewards Status

### PR DESCRIPTION
**Pull request by parv0888**
## Purpose
Added support for query `GetRewardStatus`

## Changes
Added support for query `GetRewardStatus`

## Checklist
- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

## CLA acceptance
By submitting the contribution I accept the terms and conditions of the
Contributor License Agreement v1.0
- link: https://developers.concordium.com/CLAs/Contributor-License-Agreement-v1.0.pdf

- [x] I accept the above linked CLA.

Depends On #87

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -373,6 +379,12 @@ val peers = client.getPeerList(true);
 SemVer version = client.getVersion();
 ```
 
+#### getRewardStatus
+```java
+val rewardsStatus = client
```
Comment (EmilLa1) [inline README.md:384]: Please provide type.

File: concordium-sdk/src/main/java/com/concordium/sdk/Client.java [hunk 2]
```diff
@@ -313,6 +312,25 @@ public Semver getVersion() {
         return new Semver(versionString);
     }
 
+    /**
+     * Get the information about total amount of CCD and the state of various administrative accounts.
+     * @param blockHash Block at which the reward status is to be retrieved.
+     * @return Parsed {@link RewardsOverview}.
+     * @throws Exception When the returned response is null.
+     */
+    public RewardsOverview getRewardStatus(final Hash blockHash) throws Exception {
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/Client.java:321]: Please use `BlockNotFoundException`

File: concordium-sdk/src/main/java/com/concordium/sdk/Client.java [hunk 3]
```diff
@@ -313,6 +312,25 @@ public Semver getVersion() {
         return new Semver(versionString);
     }
 
+    /**
+     * Get the information about total amount of CCD and the state of various administrative accounts.
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/Client.java:316]:
```suggestion
     * Get the information about total amount of CCD and the state of various special accounts in the provided block.
```

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/rewardstatus/RewardsOverview.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+import java.util.Date;
+
+/**
+ * Parsed Rewards Overview from response of
+ * {@link concordium.P2PGrpc.P2PBlockingStub#getRewardStatus(ConcordiumP2PRpc.BlockHash)}
+ */
+@Data
+@Getter
+public class RewardsOverview {
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/rewardstatus/RewardsOverview.java:19]: Please make fields `final` if you use `@Data`. Also then `@Getter` shouldn't be necessary.

**Review by EmilLa1 — CHANGES_REQUESTED** [review]
Please provide unit test.

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/rewardstatus/RewardsOverview.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+import com.concordium.sdk.transactions.CCDAmount;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import concordium.ConcordiumP2PRpc;
+import lombok.Data;
+import lombok.Getter;
+
+import java.util.Date;
+
+/**
+ * Parsed Rewards Overview from response of
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/rewardstatus/RewardsOverview.java:14]: This type of documentation is not really helpful for the consumer of the SDK.
Instead the documentation should state what to expect from the type and not that it's a parsed version of something else. 

E.g. 
`Get an overview of the balance of special accounts in the given block and some meta information about the state of the chain.`

File: README.md (hunk 1 above)
Comment (parv0888) [reply README.md:384]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/Client.java (hunk 2 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/Client.java:321]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/Client.java (hunk 3 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/Client.java:316]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/rewardstatus/RewardsOverview.java (hunk 4 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/rewardstatus/RewardsOverview.java:22]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/rewardstatus/RewardsOverview.java (hunk 5 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/rewardstatus/RewardsOverview.java]: have changed it  to `Information about total amount of CCD and the state of various administrative accounts.`

File: concordium-sdk/src/main/java/com/concordium/sdk/Client.java (hunk 2 above)
Comment (EmilLa1) [reply concordium-sdk/src/main/java/com/concordium/sdk/Client.java:381]: Please also reflect it in the function signature. Currently it `throws Exception` but should be `throws BlockNotFoundException` .

**Review by EmilLa1 — CHANGES_REQUESTED** [review]
Looks good. One comment about `getRewardStatus`.

**Comment by parv0888** [comment]
> 

done

**Comment by parv0888** [comment]
> Looks good. One comment about `getRewardStatus`.

done

**Comment by EmilLa1** [comment]
Closed via https://github.com/Concordium/concordium-java-sdk/pull/91

**Pull request closed without merging**